renderer/src/common/lib/meta-optimizer.ts

```diff
@@ -0,0 +1,171 @@
+import { featureFlagKeys } from '../../../../utils/feature-flags'
+import log from 'electron-log/renderer'
+import { queryClient } from './query-client'
+import {
+  getApiV1BetaGroups,
+  postApiV1BetaGroups,
+  postApiV1BetaWorkloads,
+} from '@api/sdk.gen'
+import type { V1CreateRequest } from '@api/types.gen'
+import {
+  getApiV1BetaRegistryByNameServersByServerNameOptions,
+  getApiV1BetaWorkloadsByNameOptions,
+} from '@api/@tanstack/react-query.gen'
+import { META_MCP_SERVER_NAME, MCP_OPTIMIZER_GROUP_NAME } from './constants'
+
+async function getMetaOptimizerWorkload() {
+  try {
+    const workloadDetail = await queryClient.ensureQueryData(
+      getApiV1BetaWorkloadsByNameOptions({
+        path: { name: META_MCP_SERVER_NAME },
+      })
+    )
+
+    return workloadDetail
+  } catch (error) {
+    const isNotFoundError =
+      typeof error === 'string' && error.includes('Workload not found')
+
+    if (isNotFoundError) {
+      return undefined
+    }
+
+    // Log unexpected errors
+    log.error('[getMetaOptimizerWorkload] Error fetching workload:', error)
+    return undefined
+  }
+}
+
+async function createMetaOptimizerWorkload() {
+  try {
+    const workloadDetail = await getMetaOptimizerWorkload()
+    if (workloadDetail) {
+      return workloadDetail
+    }
+
+    const { server } = await queryClient.ensureQueryData(
+      getApiV1BetaRegistryByNameServersByServerNameOptions({
+        path: {
+          name: 'default',
+          serverName: META_MCP_SERVER_NAME,
+        },
+      })
+    )
+    if (!server) {
+      log.info('[createMetaOptimizerWorkload] Server not found in the registry')
+      return
+    }
+
+    const body: V1CreateRequest = {
+      name: META_MCP_SERVER_NAME,
+      image: server.image,
+      transport: server.transport,
+      env_vars: {},
+      secrets: [],
+      cmd_arguments: [],
+      network_isolation: false,
+      volumes: [],
+      group: MCP_OPTIMIZER_GROUP_NAME,
```

Comment:
I guess here you could actually do sth like
```suggestion
      group: await ensureMetaOptimizerGroup(),
```

so basically the syntax automatically makes sure that these functions are always calledc in the correct order

Reply:
the point is that this way whenever you call the function that creates the Workload, it will inherently ensure that the group is already in place. Hence, the correct order of calling the two functions is automatically ensured & usage is simplified. 

More future proof, because the syntax itself guarantees the correct order